File: backend/ComentaApp.Domain/Entities/Complaint.cs
namespace ComentaApp.Domain.Entities
{
    public class Complaint
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public Guid CategoryId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
        public string Address { get; set; } = string.Empty;
        public ComplaintStatus Status { get; set; } = ComplaintStatus.Pending;
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        // Navigation properties
        public User User { get; set; } = null!;
        public Category Category { get; set; } = null!;
        public ICollection<ComplaintMedia> Media { get; set; } = new List<ComplaintMedia>();
        public ICollection<Like> Likes { get; set; } = new List<Like>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: frontend/src/types/complaint.ts
export interface Category {
  id: string;
  name: string;
  iconName: string;
  color: string;
}

export interface CreateComplaint {
  categoryId: string;
  title: string;
  description: string;
  latitude?: number;
  longitude?: number;
  address: string;
  mediaUrls: string[];
}

export interface Complaint {
  id: string;
  userId: string;
  user: {
    id: string;
    firstName: string;
    lastName: string;
    avatarUrl?: string;
  };
  categoryId: string;
  category: Category;
  title: string;
  description: string;
  latitude?: number;
  longitude?: number;
  address: string;
  status: string;
  media: ComplaintMedia[];
  likesCount: number;
  commentsCount: number;
  isLikedByCurrentUser: boolean;
  createdAt: string;
}

export interface ComplaintMedia {
  id: string;
  mediaUrl: string;
  mediaType: 'Image' | 'Video';
}

export interface CommentData {
  id: string;
  userId: string;
  user: {
    id: string;
    firstName: string;
    lastName: string;
    avatarUrl?: string;
  };
  complaintId: string;
  content: string;
  createdAt: string;
}

export interface CreateComment {
  content: string;
}

export interface LikeResponse {
  isLiked: boolean;
  likesCount: number;
}

export interface SearchParams {
  keyword?: string;
  categoryId?: string;
  status?: string;
  state?: string;
  city?: string;
  page?: number;
  pageSize?: number;
}

export interface PaginatedResponse<T> {
  data: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}
